### PR TITLE
makes xenos take damage when patting fire on ground

### PR DESCRIPTION
**Pull request by cuberound**
# About the pull request

patting fire on ground now slightly harms the xeno

# Explain why it's good for the game

fire patting nerfed the area denial aspect of fire greatly, watching xenomorph just extinqish fuel covered ground with few pats and walk thrue it also looks stupid. this adds MINOR damage to patting fire, making use of flames and tacticly patting them more rewarding. being harmed to save fellow sister feels better then just clicking a tile bunch with no risk at all. 

# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog
:cl:
balance: patting fire on ground harms the xeno, more intense fire means more harm
/:cl:

### CODE REVIEW

File: code/_onclick/xeno.dm [hunk 1]
```diff
@@ -54,6 +54,8 @@
 					var/turf/target_turf = target
 					for(var/obj/flamer_fire/fire in target_turf)
 						firepatted = TRUE
+						if(!(src.fire_immunity & FIRE_IMMUNITY_NO_DAMAGE))
```
Comment (kiVts) [inline code/_onclick/xeno.dm:57]:
```suggestion
						if(!(caste.fire_immunity & FIRE_IMMUNITY_NO_DAMAGE) || fire.tied_reagent.fire_penetrating)
```
essentially you dont need src, fireimmunity flags are in caste datums, and check if the fire is firepen regardless

File: code/_onclick/xeno.dm [hunk 2]
```diff
@@ -54,6 +54,8 @@
...
+							src.apply_damage(fire.burnlevel, BURN,fire)
```
Comment (kiVts) [inline code/_onclick/xeno.dm:58]:
```suggestion
							apply_damage(fire.burnlevel, BURN,fire)
```

**Comment by HeresKozmos** [comment]
Why is this necessary? Patting fire is the only reliable way outside of skills on cooldown to remove fire.

**Comment by cuberound** [comment]
> Why is this necessary? Patting fire is the only reliable way outside of skills on cooldown to remove fire.

fire is supposed to serve area denial, and 4 taps to remove it is far from that role, the damage is negligable anyway unless you are trying to put out ob flames, but that is already writen in the pr...

**Comment by Git-Nivrak** [comment]
Damage calculations to pat a single tile of fire:
Normal fire - Intensity 30 Burn Time 20: 120 damage
Green Fire - Intensity 15 Burn Time 50: 150 damage
Blue Fire - Intensity 40 Burn Time 40: 220 damage
High-Combustion Napalm - Intensity 45 Burn Time 10: 90 damage 
Sticky-Napalm: Intensity 15 Burn Time 50: 150 damage

Idk how you would consider this damage minor

**Comment by cuberound** [comment]
> Damage calculations to pat a single tile of fire: Normal fire - Intensity 30 Burn Time 20: 120 damage Green Fire - Intensity 15 Burn Time 50: 150 damage Blue Fire - Intensity 40 Burn Time 40: 220 damage High-Combustion Napalm - Intensity 45 Burn Time 10: 90 damage Sticky-Napalm: Intensity 15 Burn Time 50: 150 damage
> 
> Idk how you would consider this damage negligible by any measure

you need to take into account the duration lost in the time it takes to pat goes more like 
90
60 (you did not take into accunt that you remove 10 stacks from green flame not 5)
200 - 240
45
100 / 60 (no idea what that fuel is and it it has the armor stuff that makes it 10 duration per pat)
only blue flame goes above getting shot by a burst from rifle damage. it is negligible as long as you are removing it for others and that is what this aims for, if you cooperate one can remove flaems and other attack, but if you go in solo it is rather silly that you can pat out whole burst out of flamethrower without taking any harm

**Comment by cuberound** [comment]
and in emrgency you can still run thrue the fire and get patted on the other side, that does not harm them only you

**Comment by Red-byte3D** [comment]
with how many fire sources and the ability to use greenflame (for everyone now instead of just pyro) ontop of having big ot rocket flames this is just another useless nerf that will gimp xenos even further.

**Comment by cuberound** [comment]
> with how many fire sources and the ability to use greenflame (for everyone now instead of just pyro) ontop of having big ot rocket flames this is just another useless nerf that will gimp xenos even further. And dont forget the fact mortar incin spam exists.

as said above several times, green flame is the easiest one to remove I do not know why I have to tell it everyone individualy. mortar has limited ammo, and if you do not start imidietly but wait for a while and do not rush it drone can pat its way past seven tiles of green flame, if that feels too little the damage can be halved. but patting fire should not be free flames are supposed to be area denial, and three second slowdown of single enemy is far from that

File: code/_onclick/xeno.dm [hunk 3]
```diff
@@ -54,6 +54,8 @@
 					var/turf/target_turf = target
 					for(var/obj/flamer_fire/fire in target_turf)
 						firepatted = TRUE
+						if(!(caste.fire_immunity & FIRE_IMMUNITY_NO_DAMAGE) || fire.tied_reagent.fire_penetrating)
+							apply_damage(fire.burnlevel, BURN,fire)
```
Comment (Drulikar) [inline code/_onclick/xeno.dm:58]: Make it a prob or like greatly reduce the damage dealt.

**Review by Drulikar — CHANGES_REQUESTED** [review]

File: code/_onclick/xeno.dm (hunk 3 above)
Comment (Drulikar) [reply code/_onclick/xeno.dm:58]: Fire itself is just `burnlevel*FIRE_DAMAGE_PER_LEVEL` where FIRE_DAMAGE_PER_LEVEL is 3.

Technically it ought to use `check_fire_intensity_resistance()` too but its not really used.

File: code/_onclick/xeno.dm [hunk 4]
```diff
@@ -54,6 +54,8 @@
 					var/turf/target_turf = target
 					for(var/obj/flamer_fire/fire in target_turf)
 						firepatted = TRUE
+						if(!(caste.fire_immunity & FIRE_IMMUNITY_NO_DAMAGE) || fire.tied_reagent.fire_penetrating)
+							apply_damage(max(fire.burnlevel/2-check_fire_intensity_resistance() , 0), BURN,fire)
```
Comment (Drulikar) [inline code/_onclick/xeno.dm:58]:
```suggestion
						if(!(caste.fire_immunity & FIRE_IMMUNITY_NO_DAMAGE) || fire.tied_reagent?.fire_penetrating)
							var/firedamage = max(fire.burnlevel - check_fire_intensity_resistance(), 0) * 0.5
							apply_damage(firedamage, BURN, fire)
```

**Review by Drulikar — CHANGES_REQUESTED** [review]

**Review by Drulikar — APPROVED** [review]

**Pull request merged**